test/hotspot/jtreg/compiler/lib/ir_framework/SetupInfo.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package compiler.lib.ir_framework;
+
+/**
+ * Info optionally passed to {@link Setup} annotated methods.
+ *
+ * @see Setup
+ */
+public record SetupInfo(int invocationCounter) {
```

Comment:
I think we could also do this separately once we clean up the `AbstractInfo` class with its subclasses (I don't like how they are structured at the moment) but we should probably also add a `isWarmUp()` method at some point here such that we could use a different set of values for fields/arguments to run the compiled method if one wants to trigger an uncommon path, for example.

Reply:
We can do that. We will have to see how that plays together with automatic result verification.